### PR TITLE
browser(webkit): roll to WebKit ToT 1/29/2020

### PR DESCRIPTION
**Pull request by pavelfeldman**
~https://github.com/pavelfeldman/webkit/commit/3033a40a5a93b80d107548fea3a91daf518c3d3c~
https://github.com/pavelfeldman/webkit/commit/3941bf4475a3105fbf6434f89df92e3df20e7df6

### CODE REVIEW

File: browser_patches/webkit/patches/bootstrap.diff [hunk 1]
```diff
@@ -2719,9 +2710,12 @@ index d4e6b5d1f2151a1f4c16081b2695dd66da416ba2..7ddcea8b824043a6a17cfe2d209d861b
 +    m_frontendDispatcher->navigatedWithinDocument(frameId(&frame), url);
  }
  
--void InspectorPageAgent::didClearWindowObjectInWorld(Frame& frame)
+-void InspectorPageAgent::didClearWindowObjectInWorld(Frame& frame, DOMWrapperWorld& world)
```
Comment (yury-s) [inline browser_patches/webkit/patches/bootstrap.diff]: Why can't we use the same logic as upstream?

File: browser_patches/webkit/patches/bootstrap.diff [hunk 2]
```diff
@@ -3423,22 +3418,19 @@ index 864ce3cfe13994642495f2d7dfbdcc9a4cad9f24..50a4c389709d92a251214b24ba6068d9
...
--    notifyContextCreated(frameId, scriptState, nullptr, true);
-+    Document* document = frame.document();
-+    String worldName = document ? document->securityOrigin().toRawString() : String();
-+    notifyContextCreated(frame, scriptState, worldName, true);
-+    frame.script().evaluateIgnoringException(ScriptSourceCode(pageAgent->bootstrapScript(), URL { URL(), "web-inspector://bootstrap.js"_s }));
+     notifyContextCreated(pageAgent->frameId(&frame), frame.script().globalObject(world), world);
 +
-+    // Create user worlds.
-+    for (auto pair : pageAgent->isolatedWorldScripts())
-+        createIsolatedWorld(frame, pair.key, pair.value);
```
Comment (yury-s) [inline browser_patches/webkit/patches/bootstrap.diff]: This code used to force creation of the isolated worlds for new frames. The new code will only initialize it if already exists. We need a code that ensures isolated worlds creation for new frames.

**Review by yury-s — CHANGES_REQUESTED** [review]

File: browser_patches/webkit/patches/bootstrap.diff (hunk 1 above)
Comment (pavelfeldman) [reply browser_patches/webkit/patches/bootstrap.diff]: We should not eval bootstrap scripts in two different locations. And they should not do that either.

File: browser_patches/webkit/patches/bootstrap.diff (hunk 2 above)
Comment (pavelfeldman) [reply browser_patches/webkit/patches/bootstrap.diff]: Worlds outlive navigation, notifyCreated will be called and it'll create our isolated world.

File: browser_patches/webkit/patches/bootstrap.diff (hunk 1 above)
Comment (yury-s) [reply browser_patches/webkit/patches/bootstrap.diff]: Agree with that, you can remove 
```
if (&world != &mainThreadNormalWorld())
    return;
```
check from this method and should work for us too with a smaller diff. Do you know btw why it's not evaluated in all worlds upstream?

File: browser_patches/webkit/patches/bootstrap.diff [hunk 3]
```diff
@@ -2516,8 +2496,8 @@ index d4e6b5d1f2151a1f4c16081b2695dd66da416ba2..7ddcea8b824043a6a17cfe2d209d861b
  #include "CachedResource.h"
  #include "CachedResourceLoader.h"
  #include "Cookie.h"
-@@ -39,12 +41,15 @@
- #include "CustomHeaderFields.h"
+@@ -40,12 +42,15 @@
+ #include "DOMWrapperWorld.h"
```
Comment (yury-s) [inline browser_patches/webkit/patches/bootstrap.diff]: revert?

**Review by yury-s — APPROVED** [review]

**Pull request merged**